pkg/adaptation/plugin.go

```diff
@@ -381,17 +428,23 @@ func (p *plugin) UpdateContainers(ctx context.Context, req *UpdateContainersRequ
 }
 
 // configure the plugin and subscribe it for the events it requested.
-func (p *plugin) configure(ctx context.Context, name, version, config string) error {
+func (p *plugin) configure(ctx context.Context, name, version, config string) (err error) {
 	ctx, cancel := context.WithTimeout(ctx, getPluginRequestTimeout())
 	defer cancel()
 
-	rpl, err := p.stub.Configure(ctx, &ConfigureRequest{
+	var rpl *api.ConfigureResponse
+	req := &ConfigureRequest{
 		Config:              config,
 		RuntimeName:         name,
 		RuntimeVersion:      version,
 		RegistrationTimeout: getPluginRegistrationTimeout().Milliseconds(),
 		RequestTimeout:      getPluginRequestTimeout().Milliseconds(),
-	})
+	}
+	if p.wasm != nil {
```

Comment:
I feel like there are only two direct invoke plugin types atm but still maybe a getType func and switch for these?